core/src/main/java/com/linecorp/armeria/common/util/Unwrappable.java

```diff
@@ -87,4 +87,44 @@ default <T> T as(Class<T> type) {
     default Unwrappable unwrap() {
         return this;
     }
+
+    /**
+     * Unwraps this object and returns the innermost object being decorated.
+     * If this {@link Unwrappable} is the innermost object, this method returns itself. For example:
+     * <pre>{@code
+     * class Foo implements Unwrappable {}
+     *
+     * class Bar<T extends Unwrappable> extends AbstractUnwrappable<T> {
+     *     Bar(T delegate) {
+     *         super(delegate);
+     *     }
+     * }
+     *
+     * class Qux<T extends Unwrappable> extends AbstractUnwrappable<T> {
+     *     Qux(T delegate) {
+     *         super(delegate);
+     *     }
+     * }
+     *
+     * Foo foo = new Foo();
+     * assert foo.unwrapAll() == foo;
+     *
+     * Bar<Foo> bar = new Bar<>(foo);
+     * assert bar.unwrapAll() == foo;
+     *
+     * Qux<Bar<Foo>> qux = new Qux<>(bar);
+     * assert qux.unwrap() == bar;
+     * assert qux.unwrapAll() == foo;
+     * }</pre>
+     */
+    default Unwrappable unwrapAll() {
```

Comment:
```suggestion
    @UnstableApi
    default Unwrappable unwrapAll() {
```

Reply:
Oops 😅  Thanks for catching this 👍

Reply:
One more thing: Could we change the return type (`Unwrappable`) to `Object` in this PR?
I don't think we have to change the return type when the return type of `unwrap()` changes by #4352
Sorry, I should've left them together to handle them at once.  😅

Reply:
Thanks, done!